sandbox/host/src/sandbox/wasmer_backend.rs

```diff
@@ -133,22 +125,13 @@ pub fn invoke(
     match wasmer_result.as_ref() {
         [] => Ok(None),
 
-        [wasm_value] => {
-            let wasmer_value = match *wasm_value {
-                wasmer::Val::I32(val) => Value::I32(val),
-                wasmer::Val::I64(val) => Value::I64(val),
-                wasmer::Val::F32(val) => Value::F32(f32::to_bits(val)),
-                wasmer::Val::F64(val) => Value::F64(f64::to_bits(val)),
-                _ => {
-                    return Err(Error::Sandbox(format!(
-                        "Unsupported return value: {:?}",
-                        wasm_value,
-                    )))
-                }
-            };
-
-            Ok(Some(wasmer_value))
-        }
+        [wasm_value] => match into_value(wasm_value) {
```

Comment:
Don't you like [`ok_or_else`](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else)?

Reply:
yep, I do. However it requires to return Ok(Some(_)) if it is some not just Ok(_).